### PR TITLE
Add C implementation for `@stdlib/math/base/special/wrap`

### LINKED ISSUE #677: [RFC]: Add C implementation for `@stdlib/math/base/special/wrap`
### Description

This RFC proposes adding C implementation [`@stdlib/math/base/special/wrap`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/wrap)
```c
double stdlib_base_wrap( const double v, const double min, const double max )
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #677 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks passed, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/wrap/README.md [hunk 1]
```diff
@@ -126,6 +126,99 @@ for ( i = 0; i < 100; i++ ) {
...
+double v = stdlib_base_wrap( 3.14, 0.0, 5.0 );
+// returns 3.14
+
+v = stdlib_base_wrap( -3.14, 0.0, 5.0 );
+// returns ~1.86
+```
+
+The function accepts the following arguments:
+
+-   **v**: `[in] double`   input value to wrap.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/README.md]:
```suggestion
-   **v**: `[in] double` input value to wrap.
```

File: lib/node_modules/@stdlib/math/base/special/wrap/README.md [hunk 2]
```diff
@@ -126,6 +126,99 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double min[] = { 0.0, 0.0, 0.0, 0.0, -3.14 };
+    double max[] = { 5.0, 5.0, 5.0, 5.0, -0.0 };
+    double out;
+    double v[] = { 3.14, -3.14, 10.0, -0.0, 0.0 };
+    int i;
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/README.md]:
```suggestion
    double v[] = { 3.14, -3.14, 10.0, -0.0, 0.0 };

    double out;
    int i;
```

File: lib/node_modules/@stdlib/math/base/special/wrap/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double min[] = { 0.0, 0.0, 0.0, 0.0, -3.14 };
+    double max[] = { 5.0, 5.0, 5.0, 5.0, -0.0 };
+    double out;
+    double v[] = { 3.14, -3.14, 10.0, -0.0, 0.0 };
+    int i;
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/examples/c/example.c]:
```suggestion
    double v[] = { 3.14, -3.14, 10.0, -0.0, 0.0 };

    double out;
    int i;
```

File: lib/node_modules/@stdlib/math/base/special/wrap/include/stdlib/math/base/special/wrap.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef STDLIB_MATH_BASE_SPECIAL_WRAP_H
+#define STDLIB_MATH_BASE_SPECIAL_WRAP_H
+
+#include <stdint.h>
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/include/stdlib/math/base/special/wrap.h]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/wrap/lib/native.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Wraps a value on the half-open interval `[min,max)`.
+*
+* @param {number} v - input value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/lib/native.js:32]:
```suggestion
* @private
* @param {number} v - input value
```

File: lib/node_modules/@stdlib/math/base/special/wrap/src/main.c [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+* // returns 3.14
+*/
+double stdlib_base_wrap( const double v, const double min, const double max ) {
+	double delta;
+	double vc = v;
+	double minc = min;
+	double maxc = max;
+	if ( stdlib_base_is_nan( vc ) || stdlib_base_is_nan( minc ) || stdlib_base_is_nan( maxc ) || maxc <= minc ) {
+		return 0.0 / 0.0; // NaN
+	}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/src/main.c]:
```suggestion
	double delta;
	double maxc;
	double minc;
	double vc;

	if ( stdlib_base_is_nan( v ) || stdlib_base_is_nan( min ) || stdlib_base_is_nan( max ) || max <= min ) {
		return 0.0 / 0.0; // NaN
	}
	maxc = max;
	minc = min;
	vc = v;
```

File: lib/node_modules/@stdlib/math/base/special/wrap/src/main.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/wrap.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/trunc.h"
+#include <math.h>
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/src/main.c:23]:
```suggestion

// TODO: remove <math.h> header and fmod once we have stdlib equivalent

```

File: lib/node_modules/@stdlib/math/base/special/wrap/README.md [hunk 8]
```diff
@@ -126,6 +126,99 @@ for ( i = 0; i < 100; i++ ) {
...
+### Usage
+
+```c
+#include "stdlib/math/base/special/wrap.h"
+```
+
+#### stdlib_base_wrap( v, min, max )
+
+Wrap a value on the half-open interval `[min,max)`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/README.md]:
```suggestion
Wraps a value on the half-open interval `[min,max)`.
```

File: lib/node_modules/@stdlib/math/base/special/wrap/src/main.c [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	double minc;
+	double vc;
+
+	if ( stdlib_base_is_nan( v ) || stdlib_base_is_nan( min ) || stdlib_base_is_nan( max ) || max <= min ) {
+		return 0.0 / 0.0; // NaN
+	}
+	maxc = max;
+	minc = min;
+	vc = v;
+	// Normalize +-0 to +0...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/wrap/src/main.c:51]:
```suggestion

	// Normalize +-0 to +0...
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**